### DISCUSSION TITLE
Blurring a specific colour in multiple images?

### DISCUSSION THREAD

**Discussion by lodos3** [ImageMagick/ImageMagick#1925] [General]
Hello,

I've made a script that can pin point on a specific colour and clear it completely, which is great. However, i am trying to figure out how i can tailor the same script to do a blur on a specific pixel.

`mkdir batch

FOR %G IN (*.png) DO magick "%G" -transparent #0f0808 "batch\%G"`

Could anyone assist me on making the above script select #0f0808 and blur it ?

**Comment by snibgo** [comment]
What version of IM, on what platform? I assume v7 in a Windows BAT script.

What does "a blur on a specific pixel" mean? By definition, a blur must involve many pixels.

**Reply by lodos3** [reply]
I'm using (from what i know) the newest version of IM as i only got it installed the other day and yes im using windows BAT.

What i mean is , i want to blur only ONE colour (the one i posted above) from multiple images.

**Comment by lodos3** [comment]
Anyone? this must be doable! All i want is to blur one colour in an image, so for example if i had two triangles on a single image , one blue and one red ..only the red one would blur

**Comment by fmw42** [comment]
Post an image and we can see what can be done. Explain which color you want to blur.

**Comment by snibgo** [comment]
Sorry, I don't understand what you want. Can you supply an input image, and use Gimp or something to make a desired result, even if approximate?

Are the input images graphics (with solid colours), or photos, or what?

**Comment by lodos3** [comment]
**Before**
![BeforeEdit](https://user-images.githubusercontent.com/13706398/80231169-e66fa600-864a-11ea-970d-c454ea6dbfb6.png)

**After**
![AfterEdit](https://user-images.githubusercontent.com/13706398/80231178-e96a9680-864a-11ea-80b8-4e90241befc0.png)

The color i am aiming at blurring is **#0f0808** , i want the rest of the image intact, and this has to do be done on thousands of other images.

**Comment by fmw42** [comment]
You have a very odd image. The alpha channel is where the shadow originates. It is not in the underlying image. But it is not a unique color (gray), but actually alternating dots of black and white.

![alpha](https://user-images.githubusercontent.com/13458218/80240680-2157e680-8617-11ea-9549-233584a2445c.png)

So it is not possible to isolate it by color in order to blur only that region. One has to blur the black regions to blur the checker pattern in the shadow. But that will affect the outline of the white region.

Offhand, I do not know how to do what you want for an image of this kind.

**Comment by lodos3** [comment]
Actually the shadow isn't even opaque, therefore it doesn't it in the alpha
channel at all.

It's fine, I can do it all uses Photoshop scripts by picking the ranges of
colours that the shadows consists off.

It just takes longer and I thought image magick would make easy work of
this, clearly it's not that good and totally useless when it comes to
manipulating a single or list of colours.

**Reply by fmw42** [reply]
The shadow is in the alpha channel. Just do

convert image -alpha extract alpha.png

convert image -alpha off off.png

You will see that the shadow is black and white dots in the alpha channel (alpha.png) and not gray and that the underlying image (aoff.png) has no shadow.

If the shadow was actually gray in either image, I could give you the commands. But it is not, at least to what Imagemagick sees when it reads your image.

**Comment by snibgo** [comment]
I often approach problems by asking:

1. What effect do I want?

2. Where do I want it?

"Where" can be answered here as: in the shadow region, where the mean alpha in a sliding 3x3 window is 50%.

But I still don't know "what" is needed. Perhaps the shadow becomes a solid colour. As the shadow is partially transparent, a simple blur doesn't result in any particular colour.

**Comment by lodos3** [comment]
The shadow in the original image is not ALPHA, its solid colour, which I've
provided in my top post.

All I'm trying to do is select that shadow colour and blur it, but I dont
need any of your help anymore as I've worked it out myself.

**Reply by fmw42** [reply]
The image above that you call before.png does have the shadow in the alpha channel as a grid of black and white dots. Perhaps it got changed upon upload. Please zip your original file and post a link to that. I will look again at that.

**Comment by snibgo** [comment]
> ...  I've worked it out myself.

I encourage you to to show us what you did. This is how we all learn.
